### PR TITLE
Asus Keyboard Support

### PR DESCRIPTION
**Pull request by xQwexx**
Support Asus strix scope (tested), Asus strix flare (not tested)
Claymore (not implemented)

### CODE REVIEW

File: RGB.NET.Devices.Asus/Generic/AsusUpdateQueue.cs [hunk 1]
```diff
@@ -48,22 +48,65 @@ protected override void Update(Dictionary<object, Color> dataSet)
...
-                    light.Blue = b;
+                    foreach (KeyValuePair<object, Color> data in dataSet)
+                    {
+                        ushort index = (ushort)data.Key;
+                        IAuraSyncKeyboard keyboard = (IAuraSyncKeyboard)Device;
+                        if (keyboard != null)
+                        {
+                            IAuraRgbLight light;
+                            //UK keyboard Layout
+                            if (index == 0x56)
```
Comment (DarthAffe) [inline RGB.NET.Devices.Asus/Generic/AsusUpdateQueue.cs]: Hey, could you add some information what all these magic numbers come from and what exactly they mean?

**Comment by xQwexx** [comment]
Which specific numbers? Generally I found the numbers in the Asus sdk description (developer tutorial) but it is just worked with the us layout so I needed to map the backslash and the hashtag key to work with uk layout.

**Comment by SpoinkyNL** [comment]
@xQwexx Someone that's trying to use the Asus Strix Flare with Artemis is having some issues. Would you be able to take a look?

**Frittata_di_Maccheron on Discord:**
> Keyboard_LeftGui is mapped in editor, shows in debugger, but doesn't light up (it actually lights up LeftAlt even though it's mapped correctly in Layout editor)
Keyboard_RightGui is mapped in editor, doesn't show in debugger, there is no "Keyboard_RightGui" even listed in debugger
Keyboard_Custom36 doesn't show in LayoutEditor, but is listed only in debugger but is not mapped or colored
also LeftGui and RightGui leds either light up together with a wrong color on Artemis startup or just don't turn on

> image
> i also added a picture from the keyboard, Win and Fn keys appear yellow, but should be the same color as surrounding keys
> image

Feel free to join us at https://discord.com/invite/S3MVaC9 :)

**Comment by xQwexx** [comment]
@SpoinkyNL 
So I think I know the cause of this problem, When I wrote this pr I didn't know that the flare and scope are also the same 'Charm' name in the API, so I made following changes what is intended for the scope layout
```
	                            // Asus Strix Scope
                            if (keyboard.Name == "Charm")
                            {
                                if (index == 0XDB)
                                {
                                    light = keyboard.Lights[(int)(5 * keyboard.Width + 2)];
                                }
                                else if (index == 0x38)
                                {
                                    light = keyboard.Lights[(int)(5 * keyboard.Width + 3)];
                                }
                            }
```
What is going to be wrong in the flare layout, but because I don't own a flare I couldn't test it.
The problem that it is impossible decide which device are detected from the asus sdk itself :(

**Pull request merged**